Data/ByteString.hs

```diff
@@ -889,6 +891,22 @@ take n ps@(BS x l)
     | otherwise = BS x n
 {-# INLINE take #-}
 
+-- | /O(1)/ 'takeEnd' @n xs@ is equivalent to @'drop' ('length' xs - n) xs@.
```

Comment:
This should look a bit better.

```suggestion
-- | /O(1)/ @'takeEnd' n xs@ is equivalent to @'drop' ('length' xs - n) xs@.
```

Please apply the same fix for dropEnd and check the rendered docs.

Reply:
Would I need to also do this with all the other functions in this file?
I was just following what was already being done eg:
```
-- | /O(n)/ 'map' @f xs@
-- | /O(n)/ 'reverse' @xs@
-- | /O(1)/ 'take' @n@
```

Reply:
> Would I need to also do this with all the other functions in this file?

Oh! It would be nice if the other docs were fixed too - the spaces in these examples are very clearly _not_ monospaced. Since it's rather unrelated to the original impetus of this PR, I'd understand if you don't want to address it in this PR though.

Reply:
@sjakobi not a problem just wanted to double check. Actually what I'll do is fix this one then do a new PR looking at monospaces through out the codebase 😃 (not sure where I've got the motivation to do it but I have lol)